tools/src/main/java/org/apache/kafka/tools/RemoteLogMetadataAnalyzer.java

```diff
@@ -0,0 +1,195 @@
+package org.apache.kafka.tools;
+
+import joptsimple.OptionSpec;
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.admin.AdminClientConfig;
+import org.apache.kafka.clients.admin.TopicDescription;
+import org.apache.kafka.clients.admin.TopicListing;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.MessageFormatter;
+import org.apache.kafka.common.TopicIdPartition;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.serialization.ByteArrayDeserializer;
+import org.apache.kafka.common.utils.Exit;
+import org.apache.kafka.server.log.remote.metadata.storage.RemoteLogMetadataTopicPartitioner;
+import org.apache.kafka.server.log.remote.metadata.storage.serialization.RemoteLogMetadataSerde.RemoteLogMetadataFormatter;
+import org.apache.kafka.server.util.CommandDefaultOptions;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class RemoteLogMetadataAnalyzer {
+
+    public static void main(String[] args) {
+        RemoteLogMetadataAnalyzerOptions opts = new RemoteLogMetadataAnalyzerOptions(args);
+        try {
+            final Collection<TopicPartition> remoteLogMetadataPartition = getRemoteLogMetadataPartition(opts);
+            final KafkaConsumer<byte[], byte[]> consumer = getConsumer(opts.options.valueOf(opts.bootstrapServer));
+            consumer.assign(remoteLogMetadataPartition);
+
+            try (final MessageFormatter formatter = new RemoteLogMetadataFormatter()) {
+                final ConsumerRecords<byte[], byte[]> records = consumer.poll(Duration.ofMillis(opts.getPollDuration()));
```

Comment:
nit: splitting these nested if/for/try would improve readability

Reply:
I did it this way to avoid opening resources that wouldn't be used while not duplicating the `RemoteLogMetadataFormatter` open statement. I can fully split the two cases if that's more readable?

Reply:
I split the cases into their own functions to improve readability